.github/workflows/build-and-test.yml

```diff
@@ -339,8 +339,7 @@ jobs:
     # Sets permissions of the GITHUB_TOKEN to allow deployment to GitHub Pages
     permissions:
       id-token: write
-      pages: write
```

Comment:
I don't think we need permission for `pages`; it was added when I was experimenting with https://github.com/actions/deploy-pages, but I've moved away from it (since it requires a different deployment mechanism).